File: backend/app/models/article_model.py
from typing import Any, Dict, List

from pydantic import BaseModel

from app.models.article import Article, Section
from app.models.entity import Entity
from app.models.event import Event
from app.models.image import Image
from app.models.infobox import InfoboxField
from app.models.location import Location
from app.models.number import NumericValue
from app.models.table import Table


class NormalizedArticleModel(BaseModel):
    """Единый контракт между backend и frontend (§7 ТЗ)."""

    article: Article
    sections: List[Section] = []
    infobox: Dict[str, InfoboxField] = {}
    tables: List[Table] = []
    locations: List[Location] = []
    events: List[Event] = []
    numbers: List[NumericValue] = []
    people: List[Entity] = []
    organizations: List[Entity] = []
    works: List[Entity] = []
    relations: List[Dict[str, Any]] = []  # заглушка под 2.x, форма ещё не зафиксирована
    images: List[Image] = []
    links: List[str] = []
    metadata: Dict[str, Any] = {}

File: frontend/types/articleModel.ts
import { Article, Section } from "./article";
import { Entity } from "./entity";
import { Event } from "./event";
import { Image } from "./image";
import { InfoboxField } from "./infobox";
import { Location } from "./location";
import { NumericValue } from "./number";
import { Table } from "./table";

export interface NormalizedArticleModel {
  article: Article;
  sections: Section[];
  infobox: Record<string, InfoboxField>;
  tables: Table[];
  locations: Location[];
  events: Event[];
  numbers: NumericValue[];
  people: Entity[];
  organizations: Entity[];
  works: Entity[];
  relations: Record<string, unknown>[];
  images: Image[];
  links: string[];
  metadata: Record<string, unknown>;
}